grpc/server.go

```diff
@@ -0,0 +1,136 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+
+	v1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	grpc_calcapi "github.com/DataDog/chaos-controller/grpc/calculations"
+	pb "github.com/DataDog/chaos-controller/grpc/disruption_listener"
+	"github.com/DataDog/chaos-controller/log"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+/*
+ * When a random integer from 0 to 100 is randomly selected, the PercentToAlteration mapping is referenced to
+ * identify the disruption to apply to a query. The mapping represents user preference the proportion of queries
+ * affected by each alteration. See docs/grpc_disruption/interceptor_algorithm.md for examples.
+ */
+
+// DisruptionListener is a gRPC Service that can disrupt endpoints of a gRPC server.
```

Comment:
The package structure under `grpc` is a bit confusing to me. Why isn't `DisruptionListener` in the `disruption_listener` package?